From c4ecd1ca2722b9f6a7344506ded7fa8967630b68 Mon Sep 17 00:00:00 2001
From: Konstantin Porotchkin <kostap@marvell.com>
Date: Sun, 4 Apr 2021 14:06:09 +0300
Subject: [PATCH 1409/1921] arm64: marvell: add pcie-ep related nodes to cp11x
 dtsi

The following DTSI nodes added for PCIe End Point mode:
- armada_ep: used by PCIe End Point facility management
  framework
- pcie_ep: defines the PCIe EP address space
- pci_ep_uio: defines the PCIe EP BARs and host mapping

Signed-off-by: Konstantin Porotchkin <kostap@marvell.com>
Change-Id: I6bb7a44edca767f6123f5dfe5fce8b7f37f4321e
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/49161
Tested-by: Stefan Chulski <stefanc@marvell.com>
Reviewed-by: Stefan Chulski <stefanc@marvell.com>
Reviewed-by: Ofer Heifetz <oferh@marvell.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-cp11x.dtsi | 27 +++++++++++++++++++
 1 file changed, 27 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-cp11x.dtsi b/arch/arm64/boot/dts/marvell/armada-cp11x.dtsi
index 249dbe764559..1d0b02af88e7 100644
--- a/arch/arm64/boot/dts/marvell/armada-cp11x.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-cp11x.dtsi
@@ -11,6 +11,7 @@
 #include "armada-common.dtsi"
 
 #define CP11X_PCIEx_CONF_BASE(iface)	(CP11X_PCIEx_MEM_BASE(iface) + CP11X_PCIEx_MEM_SIZE(iface))
+#define CP11X_PCIEx_EP_REG_BASE(iface)	(ADDRESSIFY(CP11X_BASE) + 0x600000 + (iface) * 0x4000)
 
 / {
 	/*
@@ -548,6 +549,32 @@
 		};
 	};
 
+	CP11X_LABEL(armada_ep):armada-ep {
+		compatible = "marvell,armada-ep";
+		msi-parent = <&gic_v2m0>;
+		status = "disabled";
+	};
+
+	CP11X_LABEL(pcie_ep): pcie-ep@600000 {
+		compatible = "marvell,armada-pcie-ep", "snps,dw-pcie";
+		reg =	<0 CP11X_PCIEx_EP_REG_BASE(0) 0 0x4000>,
+			<0 CP11X_PCIEx_EP_REG_BASE(2) 0 0x80000>,
+			<0 CP11X_PCIEx_EP_REG_BASE(1) 0 0x1000>;
+		reg-names = "core", "lm", "shadow_core";
+		clocks = <&CP11X_LABEL(clk) 1 13>;
+		status = "disabled";
+	};
+
+	CP11X_LABEL(pci_ep_uio): pci-ep-uio {
+		compatible = "marvell,pci-ep-uio";
+		reg = <0x00 0x00000000 0x0 0x00100000>,
+		      <0x00 0x3f000000 0x0 0x01000000>,
+		      <0x00 0xf0000000 0x0 0x01000000>,
+		      <0x80 0x00000000 0x8 0x00000000>;
+		reg-names = "bar0", "bar2", "bar4", "host-map";
+		status = "disabled";
+	};
+
 	CP11X_LABEL(pcie0): pcie@CP11X_PCIE0_BASE {
 		compatible = "marvell,armada8k-pcie", "snps,dw-pcie";
 		reg = <0 ADDRESSIFY(CP11X_PCIE0_BASE) 0 0x10000>,
-- 
2.31.1

